src/local_pathfinding/land/generate_land_data.py

```diff
@@ -0,0 +1,70 @@
+import csv
+import os
+import pickle
+from os.path import normpath
+from typing import Any
+
+import geopandas as gpd
+import pyproj
+from shapely.geometry import LineString, MultiPolygon, Polygon
+from shapely.ops import split
+
+
+def dump_pkl(object: Any, file_path: str):
+    os.makedirs(os.path.dirname(file_path), exist_ok=True)
+    with open(file_path, "wb") as f:
+        pickle.dump(object, f, protocol=pickle.HIGHEST_PROTOCOL)
+
+
+def load_pkl(file_path: str) -> Any:
+    with open(file_path, "rb") as f:
+        return pickle.load(f)
+
+
+def pickle_land():
+
+    WGS84 = pyproj.CRS("EPSG:4326")
+    MERCATOR = pyproj.CRS("EPSG:3857")
+    INLAND_POLYGON = normpath(
+        "/workspaces/sailbot_workspace/src/local_pathfinding/land/csv/map_sel.csv"
+    )
+    BUFFER_AMOUNT = 5000  # meters
+
+    gdf = gpd.read_file(
+        "/workspaces/sailbot_workspace/src/local_pathfinding/land/territorial_waters.geojson"
+    )
+
+    # the only useful data is in the id and geometry columns and in the first two rows
+    gdf = gdf[["id", "geometry"]]
+    gdf = gdf.head(2)
+    geoms = list(gdf.iloc[0]["geometry"].geoms) + list(gdf.iloc[1]["geometry"].geoms)
+
+    # convert to MultiPolygon for easier geometric manipulation
+    multi_poly = MultiPolygon(geoms)
+    multi_poly = multi_poly.buffer(0)  # repairs invalid geometry
+    # this is to prevent geometry the crosses -180 longitude from being distorted by transforms
+    with open(INLAND_POLYGON, "r") as f:
```

Comment:
Could add error handling here

Reply:
Sure! which errors do you think we should plan for here?

Reply:
The file cannot open exception error